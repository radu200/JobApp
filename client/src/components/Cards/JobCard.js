import React from 'react'
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import { JobDetailsUrl} from '../Utils/Paths/UrlPaths'
import { Link } from 'react-router-dom'
const styles = {
  JobDetails:{
    textDecoration:'none',
    color:'blue'
  },

  media: {
    // ⚠️ object-fit is not supported by IE 11.
    objectFit: 'cover',
  },

  heading:{
    opacity:0.7
  }
};
 const salaryCurrency = 'Lei'

  
const JobCard = ({jobs,classes}) => {
  
  return jobs.map((job,index) => {

     const jobDetailPath = `${JobDetailsUrl}${job.id}`

      return(
              
              <Grid  key={index}item xs={12} sm={6} md={4}>
                <Card className={classes.card} >
                  <CardActionArea>
                  {job.image ? (
                  <CardMedia
                    component="img"
                    alt={job.category}
                    className={classes.media}
                    height="140"
                    image={job.image}
                    title={job.category}
                    />
                     ) : (
                    null
                  )}
                  <CardContent>
                  <Typography    gutterBottom component="p" color="primary">
                       {job.salary ? `${job.salary} ${salaryCurrency}` : null}
                    </Typography>
                    <Typography component="p">
                       {job.employment_type}
                    </Typography>
                    <Typography gutterBottom variant="h5" component="h2">
                    {job.position}
                    </Typography>
                    <Typography component="p">
                       {job.category}
                    </Typography>
                    <Typography  noWrap className={classes.heading} component="p">
                       {job.description}
                    </Typography>
                    <Typography component="p">
                          {job.city}
                    </Typography>
                  </CardContent>
                </CardActionArea>
                    <CardActions>
                    <Button size="small" color="primary">
                        <a className={classes.JobDetails} href={jobDetailPath}>Vezi Mai Mult</a>
                    </Button>
    
                  </CardActions>
                </Card>
           </Grid>
    
        
           )
      } )

}

JobCard.propTypes = {
     classes: PropTypes.object.isRequired,
   };

export default withStyles(styles)(JobCard);