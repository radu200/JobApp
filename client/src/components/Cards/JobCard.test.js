import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
configure({adapter: new Adapter()});
import JobCard from './JobCard'

describe('CandidateCard',() => {
    it('should render correctly with no props', () => {
        const component = shallow(<JobCard />)
        expect(component).toMatchSnapshot();

    })
    
it('should render banner text correctly with given strings', () => {
    const strings = ['one', 'two'];
    const component = shallow(<JobCard list={strings} />);
    expect(component).toMatchSnapshot();
    });
})