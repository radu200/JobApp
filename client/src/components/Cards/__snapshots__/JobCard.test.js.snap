// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CandidateCard should render banner text correctly with given strings 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef(WithStyles)
    list={
      Array [
        "one",
        "two",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "JobDetails": "JobCard-JobDetails-1",
        "heading": "JobCard-heading-3",
        "media": "JobCard-media-2",
      },
      "list": Array [
        "one",
        "two",
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "JobDetails": "JobCard-JobDetails-1",
          "heading": "JobCard-heading-3",
          "media": "JobCard-media-2",
        },
        "list": Array [
          "one",
          "two",
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`CandidateCard should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef(WithStyles) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "JobDetails": "JobCard-JobDetails-1",
        "heading": "JobCard-heading-3",
        "media": "JobCard-media-2",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "JobDetails": "JobCard-JobDetails-1",
          "heading": "JobCard-heading-3",
          "media": "JobCard-media-2",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
